mapie/regression.py

```diff
@@ -488,6 +481,8 @@ def fit(
         y = _check_y(y)
         self.n_features_in_ = check_n_features_in(X, cv, estimator)
         sample_weight, X, y = check_null_weight(sample_weight, X, y)
+        y = cast(NDArray, y)
```

Comment:
Shouldn't we also cast `X` so all methods accept and return only `NDArray` arguments ?

Reply:
Alas no, because as some users noticed in this issue https://github.com/scikit-learn-contrib/MAPIE/issues/128, MAPIE should not convert input arrays as numpy arrays on its own, but should let the wrapped model manage arbitrary inputs. One example is a wrapped model being a pipeline treating pandas dataframe with a scikit-learn ColumnTransformer : the latter needs input to be pandas dataframe in order to record column names. Would MAPIE convert this to a numpy array, column names would be lost, andthe compatibility with such pipelines would be lost.